DigitalLearningSolutions.Web/Scripts/searchSortFilterAndPaginate/searchSortFilterAndPaginate.ts

```diff
@@ -309,24 +321,31 @@ export class SearchSortFilterAndPaginate {
       }
     });
 
-    return keptQueryParameters.length > 0 ? `?${keptQueryParameters.join('&')}` : '';
+    const baseQuery = `?${SearchSortFilterAndPaginate.getBaseQueryParameters()}`;
+    return keptQueryParameters.length > 0 ? `${baseQuery}&${keptQueryParameters.join('&')}` : baseQuery;
   }
 
-  private updateSearchableElementLinks(searchableData: ISearchableData): void {
-    const searchBoxContent = getQuery();
-    const setReturnPage = !this.searchEnabled
-      || (searchBoxContent != null && searchBoxContent.length === 0);
+  private static getBaseQueryParameters(): string {
+    const searchString = getQuery();
+    const sortBy = getSortBy();
+    const sortDirection = getSortDirection();
+    const itemsPerPage = getItemsPerPageValue().toString();
+    return `searchString=${searchString}&sortBy=${sortBy}&sortDirection=${sortDirection}&itemsPerPage=${itemsPerPage}`;

```

Comment:
I can't really think of anything cleaner than adding a sortEnabled and itemsPerPageEnabled bool to the constructor and using these. I think it would make the no JS side much more of a pain though, as I'd have to modify the ReturnPageQuery.ToString to also do something similar, but we have no information there as to what is enabled.